bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/things/AutomowerHandler.java

```diff
@@ -0,0 +1,285 @@
+        super(thing);
+    }
+
+    class AutomowerPollingRunnable implements Runnable {
+
+        @Override
+        public void run() {
+            Bridge bridge = getBridge();
+            if (bridge != null && bridge.getStatus() == ThingStatus.ONLINE) {
+                updateAutomowerState();
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            refreshChannels(channelUID);
+        }
+    }
+
+    private void refreshChannels(ChannelUID channelUID) {
+        updateAutomowerState();
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(AutomowerActions.class);
+    }
+
+    @Override
+    public void initialize() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            AutomowerConfiguration currentConfig = getConfigAs(AutomowerConfiguration.class);
+            final String configMowerId = currentConfig.getMowerId();
+            final Integer pollingIntervalS = currentConfig.getPollingInterval();
+
+            if (configMowerId == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/conf-error-no-mower-id");
+            } else if (pollingIntervalS != null && pollingIntervalS < 1) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/conf-error-invalid-polling-interval");
+            } else {
+                automowerId.set(configMowerId);
+                startAutomowerPolling(pollingIntervalS);
+            }
+
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
+        }
+    }
+
+    @Nullable
+    private AutomowerBridge getAutomowerBridge() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            ThingHandler handler = bridge.getHandler();
+            if (handler instanceof AutomowerBridgeHandler) {
+                AutomowerBridgeHandler bridgeHandler = (AutomowerBridgeHandler) handler;
+                return bridgeHandler.getAutomowerBridge();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void dispose() {
+        if (!automowerId.get().equals(NO_ID)) {
+            stopAutomowerPolling();
+            automowerId.set(NO_ID);
+        }
+    }
+
+    private void startAutomowerPolling(@Nullable Integer pollingIntervalS) {
+        if (automowerPollingJob == null) {
+            final long pollingIntervalToUse = pollingIntervalS == null ? DEFAULT_POLLING_INTERVAL_S : pollingIntervalS;
+            automowerPollingJob = scheduler.scheduleWithFixedDelay(new AutomowerPollingRunnable(), 1,
```

Comment:
Instead of a nested class, could you just make `AutomowerPollingRunnable` a lambda instead?

Reply:
yes